### PR TITLE
chore: adaptations for nightly-2024-08-28

### PR DESCRIPTION
**Pull request by jcommelin**

### CODE REVIEW

File: Mathlib/Algebra/Lie/DirectSum.lean [hunk 1]
```diff
@@ -229,10 +229,13 @@ variable {L : Type w} [LieRing L] [LieAlgebra R L] (I : ι → LieIdeal R L)
 [this Zulip thread](https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/
 Typeclass.20resolution.20under.20binders/near/245151099). -/
 instance lieRingOfIdeals : LieRing (⨁ i, I i) :=
+  letI : (i : ι) → LieRing (I i) := fun _ => LieIdeal.lieRing ..
```
Comment (jcommelin) [inline Mathlib/Algebra/Lie/DirectSum.lean:232]:
```suggestion
  #adaptation_note /-- After lean4#5020, many instances for Lie algebras and manifolds are no
  longer found. -/
  letI : (i : ι) → LieRing (I i) := fun _ => LieIdeal.lieRing ..
```

File: Mathlib/Algebra/Lie/DirectSum.lean [hunk 2]
```diff
@@ -229,10 +229,13 @@ variable {L : Type w} [LieRing L] [LieAlgebra R L] (I : ι → LieIdeal R L)
...
   DirectSum.lieRing fun i => ↥(I i)
 
 /-- See `DirectSum.lieRingOfIdeals` comment. -/
 instance lieAlgebraOfIdeals : LieAlgebra R (⨁ i, I i) :=
+  letI : (i : ι) → LieAlgebra R (I i) := fun _ => LieIdeal.lieAlgebra ..
```
Comment (jcommelin) [inline Mathlib/Algebra/Lie/DirectSum.lean:237]:
```suggestion
  #adaptation_note /-- After lean4#5020, many instances for Lie algebras and manifolds are no
  longer found. -/
  letI : (i : ι) → LieAlgebra R (I i) := fun _ => LieIdeal.lieAlgebra ..
```

File: Mathlib/Algebra/Lie/EngelSubalgebra.lean [hunk 3]
```diff
@@ -113,6 +113,7 @@ lemma normalizer_eq_self_of_engel_le [IsArtinian R L]
     rwa [← lie_skew, neg_mem_iff (G := L)]
   have aux₂ : ∀ n ∈ N, ⁅x, n⁆ ∈ N := fun n hn ↦ le_normalizer H (aux₁ _ hn)
   let dx : N →ₗ[R] N := (ad R L x).restrict aux₂
+  have : IsArtinian R { x // x ∈ N } := isArtinian_submodule' _
```
Comment (jcommelin) [inline Mathlib/Algebra/Lie/EngelSubalgebra.lean:116]: Why is this no longer found automatically?

File: Mathlib/Algebra/Lie/IdealOperations.lean [hunk 4]
```diff
@@ -213,6 +213,7 @@ theorem map_bracket_eq [LieModule R L M] : map f ⁅I, N⁆ = ⁅I, map f N⁆ :
     exact ⟨x, ⟨f n, (mem_map (f n)).mpr ⟨n, hn, rfl⟩⟩, hm⟩
   · rintro ⟨x, ⟨m₂, hm₂ : m₂ ∈ map f N⟩, rfl⟩
     obtain ⟨n, hn, rfl⟩ := (mem_map m₂).mp hm₂
+    letI : Bracket I M := LieRingModule.toBracket
```
Comment (jcommelin) [inline Mathlib/Algebra/Lie/IdealOperations.lean:216]:
```suggestion
    #adaptation_note /-- After lean4#5020, many instances for Lie algebras and manifolds are no
    longer found. -/
    letI : Bracket I M := LieRingModule.toBracket
```

File: Mathlib/Algebra/Lie/Nilpotent.lean [hunk 5]
```diff
@@ -412,6 +412,8 @@ lemma disjoint_lowerCentralSeries_maxTrivSubmodule_iff [IsNilpotent R L M] :
   suffices ¬ Nontrivial (lowerCentralSeriesLast R L M) by
     exact this (nontrivial_lowerCentralSeriesLast R L M)
   rw [h.eq_bot, le_bot_iff] at this
+  letI unique : Unique (⊥ : LieSubmodule R L M) := Submodule.uniqueBot
```
Comment (jcommelin) [inline Mathlib/Algebra/Lie/Nilpotent.lean:415]:
```suggestion
  #adaptation_note /-- After lean4#5020, many instances for Lie algebras and manifolds are no
  longer found. -/
  letI unique : Unique (⊥ : LieSubmodule R L M) := Submodule.uniqueBot
```

File: Mathlib/Algebra/Lie/Nilpotent.lean [hunk 6]
```diff
@@ -588,7 +590,13 @@ theorem LieModule.isNilpotent_of_top_iff :
   Equiv.lieModule_isNilpotent_iff LieSubalgebra.topEquiv (1 : M ≃ₗ[R] M) fun _ _ => rfl
 
 @[simp] lemma LieModule.isNilpotent_of_top_iff' :
+    letI : LieRingModule L (⊤ : LieSubmodule R L M) :=
```
Comment (jcommelin) [inline Mathlib/Algebra/Lie/Nilpotent.lean:593]:
```suggestion
    #adaptation_note /-- After lean4#5020, many instances for Lie algebras and manifolds are no
    longer found. -/
    letI : LieRingModule L (⊤ : LieSubmodule R L M) :=
```
I don't know if this works. Maybe it has to be an ordinary comment. If so, I suggest leaving the string `#adaptation_note` inside the comment.

File: Mathlib/Algebra/Lie/Semisimple/Basic.lean [hunk 7]
```diff
@@ -301,7 +302,10 @@ theorem subsingleton_of_hasTrivialRadical_lie_abelian [HasTrivialRadical R L] [h
 
 theorem abelian_radical_of_hasTrivialRadical [HasTrivialRadical R L] :
     IsLieAbelian (radical R L) := by
-  rw [HasTrivialRadical.radical_eq_bot]; infer_instance
+  letI : Unique (⊥ : LieIdeal R L) := Submodule.uniqueBot
```
Comment (jcommelin) [inline Mathlib/Algebra/Lie/Semisimple/Basic.lean:305]:
```suggestion
  #adaptation_note /-- After lean4#5020, many instances for Lie algebras and manifolds are no
  longer found. -/
  letI : Unique (⊥ : LieIdeal R L) := Submodule.uniqueBot
```

File: Mathlib/Algebra/Lie/Solvable.lean [hunk 8]
```diff
@@ -287,7 +287,7 @@ instance [IsSolvable R L] : IsSolvable R (⊤ : LieSubalgebra R L) := by
 
 @[simp] lemma radical_eq_top_of_isSolvable [IsSolvable R L] :
     radical R L = ⊤ := by
-  rw [eq_top_iff]; exact le_sSup <| inferInstanceAs (IsSolvable R (⊤ : LieIdeal R L))
+  rw [eq_top_iff]; exact le_sSup <| LieAlgebra.instIsSolvableSubtypeMemLieSubalgebraTop R L
```
Comment (jcommelin) [inline Mathlib/Algebra/Lie/Solvable.lean:290]:
```suggestion
  #adaptation_note /-- After lean4#5020, many instances for Lie algebras and manifolds are no
  longer found. -/
  rw [eq_top_iff]; exact le_sSup <| LieAlgebra.instIsSolvableSubtypeMemLieSubalgebraTop R L
```

File: Mathlib/Algebra/Lie/TraceForm.lean [hunk 9]
```diff
@@ -324,6 +324,7 @@ lemma traceForm_eq_zero_of_isTrivial [LieModule.IsTrivial I N] :
     simp [this, N.trace_eq_trace_restrict_of_le_idealizer I h x hy]
   ext n
   suffices ⁅y, (n : M)⁆ = 0 by simp [this]
+  letI : Bracket I N := LieRingModule.toBracket
```
Comment (jcommelin) [inline Mathlib/Algebra/Lie/TraceForm.lean:327]:
```suggestion
  #adaptation_note /-- After lean4#5020, many instances for Lie algebras and manifolds are no
  longer found. -/
  letI : Bracket I N := LieRingModule.toBracket
```

File: Mathlib/Algebra/Lie/Weights/Basic.lean [hunk 10]
```diff
@@ -219,6 +219,8 @@ instance [Subsingleton M] : IsEmpty (Weight R L M) :=
   ⟨fun h ↦ h.2 (Subsingleton.elim _ _)⟩
 
 instance [Nontrivial (genWeightSpace M (0 : L → R))] : Zero (Weight R L M) :=
+  letI : Unique (⊥ : LieSubmodule R L M) := Submodule.uniqueBot
```
Comment (jcommelin) [inline Mathlib/Algebra/Lie/Weights/Basic.lean:222]:
```suggestion
  #adaptation_note /-- After lean4#5020, many instances for Lie algebras and manifolds are no
  longer found. -/
  letI : Unique (⊥ : LieSubmodule R L M) := Submodule.uniqueBot
```

File: Mathlib/Algebra/Lie/Weights/Basic.lean [hunk 11]
```diff
@@ -583,7 +585,7 @@ private lemma isCompl_genWeightSpace_zero_posFittingComp_aux
   · suffices IsNilpotent R L M by simp [M₀, M₁, isCompl_top_bot]
     replace h : M₀ = ⊤ := by simpa [M₀, genWeightSpace]
     rw [← LieModule.isNilpotent_of_top_iff', ← h]
-    infer_instance
+    exact LieModule.instIsNilpotentSubtypeMemSubmoduleGenWeightSpaceOfNatForallOfIsNoetherian M
```
Comment (jcommelin) [inline Mathlib/Algebra/Lie/Weights/Basic.lean:588]:
```suggestion
    #adaptation_note /-- After lean4#5020, many instances for Lie algebras and manifolds are no
    longer found. -/
    exact LieModule.instIsNilpotentSubtypeMemSubmoduleGenWeightSpaceOfNatForallOfIsNoetherian M
```

File: Mathlib/Combinatorics/SimpleGraph/Matching.lean [hunk 12]
```diff
@@ -198,7 +198,8 @@ variable [Fintype V]
 lemma even_card_of_isPerfectMatching [DecidableEq V] [DecidableRel G.Adj]
     (c : ConnectedComponent G) (hM : M.IsPerfectMatching) :
     Even (Fintype.card c.supp) := by
-  classical simpa using (hM.induce_connectedComponent_isMatching c).even_card
+  letI : DecidablePred fun x ↦ x ∈ (M.induce c.supp).verts := fun a ↦ G.instDecidableMemSupp c a
```
Comment (jcommelin) [inline Mathlib/Combinatorics/SimpleGraph/Matching.lean:201]: Is this the same "issue" as in the Lie/manifold case?
```suggestion
  #adaptation_note /-- After lean4#5020, many instances for Lie algebras and manifolds are no
  longer found. -/
  letI : DecidablePred fun x ↦ x ∈ (M.induce c.supp).verts := fun a ↦ G.instDecidableMemSupp c a
```

File: Mathlib/Geometry/Manifold/ChartedSpace.lean [hunk 13]
```diff
@@ -1161,6 +1161,7 @@ theorem chartAt_subtype_val_symm_eventuallyEq (U : Opens M) {x : U} :
   exact Filter.eventuallyEq_of_mem heUx_nhds (e.subtypeRestr_symm_eqOn ⟨x⟩)
 
 theorem chartAt_inclusion_symm_eventuallyEq {U V : Opens M} (hUV : U ≤ V) {x : U} :
+    letI : ChartedSpace H V := V.instChartedSpace
```
Comment (jcommelin) [inline Mathlib/Geometry/Manifold/ChartedSpace.lean:1164]: Not sure if `#adaptation_note` works here. Maybe needs to be regular comment.
```suggestion
    #adaptation_note /-- After lean4#5020, many instances for Lie algebras and manifolds are no
    longer found. -/
    letI : ChartedSpace H V := V.instChartedSpace
```

File: Mathlib/Geometry/Manifold/ContMDiff/Basic.lean [hunk 14]
```diff
@@ -322,11 +322,14 @@ theorem ContMDiff.extend_one [T2Space M] [One M'] {n : ℕ∞} {U : Opens M} {f
...
   lift x to U using Subtype.coe_image_subset _ _
     (supp.mulTSupport_extend_one_subset continuous_subtype_val h)
-  rw [← contMdiffAt_subtype_iff, ← comp_def]
+  rw [← contMdiffAt_subtype_iff]
+  simp_rw [← comp_def]
   erw [extend_comp Subtype.val_injective]
   exact diff.contMDiffAt
 
 theorem contMDiff_inclusion {n : ℕ∞} {U V : Opens M} (h : U ≤ V) :
+    letI : ChartedSpace H U := U.instChartedSpace
```
Comment (jcommelin) [inline Mathlib/Geometry/Manifold/ContMDiff/Basic.lean:331]:
```suggestion
    #adaptation_note /-- After lean4#5020, many instances for Lie algebras and manifolds are no
    longer found. -/
    letI : ChartedSpace H U := U.instChartedSpace
```
maybe needs to be regular comment

File: Mathlib/Geometry/Manifold/ContMDiff/Basic.lean [hunk 15]
```diff
@@ -345,7 +348,10 @@ theorem Smooth.extend_one [T2Space M] [One M'] {U : Opens M} {f : U → M'}
     (supp : HasCompactMulSupport f) (diff : Smooth I I' f) : Smooth I I' (Subtype.val.extend f 1) :=
   ContMDiff.extend_one supp diff
 
-theorem smooth_inclusion {U V : Opens M} (h : U ≤ V) : Smooth I I (Set.inclusion h : U → V) :=
+theorem smooth_inclusion {U V : Opens M} (h : U ≤ V) :
+    letI : ChartedSpace H U := U.instChartedSpace
```
Comment (jcommelin) [inline Mathlib/Geometry/Manifold/ContMDiff/Basic.lean:352]:
```suggestion
    #adaptation_note /-- After lean4#5020, many instances for Lie algebras and manifolds are no
    longer found. -/
    letI : ChartedSpace H U := U.instChartedSpace
```
maybe needs to be regular comment

File: Mathlib/Geometry/Manifold/LocalInvariantProperties.lean [hunk 16]
```diff
@@ -515,7 +515,11 @@ theorem liftProp_subtype_val {Q : (H → H) → Set H → H → Prop} (hG : Loca
 
 theorem liftProp_inclusion {Q : (H → H) → Set H → H → Prop} (hG : LocalInvariantProp G G Q)
     (hQ : ∀ y, Q id univ y) {U V : Opens M} (hUV : U ≤ V) :
+    letI : ChartedSpace H { x // x ∈ V } := V.instChartedSpace
+    letI : ChartedSpace H { x // x ∈ U } := U.instChartedSpace
     LiftProp Q (Set.inclusion hUV : U → V) := by
+  letI : ChartedSpace H { x // x ∈ V } := V.instChartedSpace
```
Comment (jcommelin) [inline Mathlib/Geometry/Manifold/LocalInvariantProperties.lean:521]:
```suggestion
  #adaptation_note /-- After lean4#5020, many instances for Lie algebras and manifolds are no
  longer found (also in the statement of this theorem). -/
  letI : ChartedSpace H { x // x ∈ V } := V.instChartedSpace
```

File: Mathlib/Geometry/Manifold/Sheaf/Basic.lean [hunk 17]
```diff
@@ -55,6 +55,7 @@ def StructureGroupoid.LocalInvariantProp.localPredicate (hG : LocalInvariantProp
   res := by
     intro U V i f h x
     have hUV : U ≤ V := CategoryTheory.leOfHom i
+    letI : ChartedSpace H U := U.instChartedSpace
```
Comment (jcommelin) [inline Mathlib/Geometry/Manifold/Sheaf/Basic.lean:58]:
```suggestion
    #adaptation_note /-- After lean4#5020, many instances for Lie algebras and manifolds are no
    longer found. -/
    letI : ChartedSpace H U := U.instChartedSpace
```

File: Mathlib/Geometry/Manifold/Sheaf/LocallyRingedSpace.lean [hunk 18]
```diff
@@ -94,6 +94,7 @@ theorem smoothSheafCommRing.isUnit_stalk_iff {x : M}
         apply inv_mul_cancel₀
         exact hVf y
     · intro y
+      letI : ChartedSpace HM V := V.instChartedSpace
```
Comment (jcommelin) [inline Mathlib/Geometry/Manifold/Sheaf/LocallyRingedSpace.lean:97]:
```suggestion
      #adaptation_note /-- After lean4#5020, many instances for Lie algebras and manifolds are no
      longer found. -/
      letI : ChartedSpace HM V := V.instChartedSpace
```

File: Mathlib/GroupTheory/SpecificGroups/Cyclic.lean [hunk 19]
```diff
@@ -109,6 +109,7 @@ theorem isCyclic_of_orderOf_eq_card [Fintype α] (x : α) (hx : orderOf x = Fint
   use x
   rw [← Set.range_iff_surjective, ← coe_zpowers]
   rw [← Fintype.card_congr (Equiv.Set.univ α), ← Fintype.card_zpowers] at hx
+  letI : Fintype (zpowers x) := (Subgroup.zpowers x).instFintypeSubtypeMemOfDecidablePred
```
Comment (jcommelin) [inline Mathlib/GroupTheory/SpecificGroups/Cyclic.lean:112]: Is this the same issue?
```suggestion
  #adaptation_note /-- After lean4#5020, many instances for Lie algebras and manifolds are no
  longer found. -/
  letI : Fintype (zpowers x) := (Subgroup.zpowers x).instFintypeSubtypeMemOfDecidablePred
```

File: Mathlib/Topology/MetricSpace/Polish.lean [hunk 20]
```diff
@@ -328,7 +328,8 @@ theorem _root_.IsOpen.polishSpace {α : Type*} [TopologicalSpace α] [PolishSpac
     (hs : IsOpen s) : PolishSpace s := by
   letI := upgradePolishSpace α
   lift s to Opens α using hs
-  have : SecondCountableTopology s.CompleteCopy := inferInstanceAs (SecondCountableTopology s)
+  have : SecondCountableTopology s.CompleteCopy :=
```
Comment (jcommelin) [inline Mathlib/Topology/MetricSpace/Polish.lean:331]: Is this the same issue?
```suggestion
  #adaptation_note /-- After lean4#5020, many instances for Lie algebras and manifolds are no
  longer found. -/
  have : SecondCountableTopology s.CompleteCopy :=
```

File: Mathlib/Algebra/Lie/EngelSubalgebra.lean (hunk 3 above)
Comment (semorrison) [inline Mathlib/Algebra/Lie/EngelSubalgebra.lean:116]:
```suggestion
  #adaptation_note /-- After lean4#5020, many instances are no longer found. -/
  have : IsArtinian R { x // x ∈ N } := isArtinian_submodule' _
```

**Pull request merged**